src/js/store/listeners.js

```diff
@@ -27,7 +27,7 @@ var sg = require('../model/signals'),
     parseInProgress = require('../actions/vegaParse');
 
 function instantiatePrimitivesFromStore(store, model) {
-  store.getState().get('primitives').forEach(function(props, markId) {
+  getIn(store.getState(), 'marks').forEach(function(props, markId) {
```

Comment:
Hey @arvind , when future people join the project who are new to redux & immutable.js, make sure you explain the what the immutable utility is.  That way when they look at the code and see the getIn()  for one level fetches from the store eg. `store.getState().get('mapname')` they don't get confused about it not matching the immutable docs.